core/src/main/scala/org/http4s/Service.scala

```diff
@@ -2,32 +2,372 @@ package org.http4s
 
 import scalaz.Kleisli
 import scalaz.concurrent.Task
-import scalaz.syntax.kleisli._
+
+final case class Service[A, B](run: PartialFunction[A, Task[B]]) {
+  import Service._
+
+  def apply(a: A)(implicit ev: Task[Response] =:= Task[B]): Task[B] =
+    applyOrElse(a, (_: A) => ev(HttpService.notFound))
```

Comment:
Scalaz has Liskov for this, whose concrete improvements over the default evidence types elude me.  I'd like to be educated on this.